cpp/ixtheo_notation_stats.cc

```diff
@@ -0,0 +1,157 @@
+/** \file    ixtheo_notation_stats.cc
+ *  \brief   Gather statistics about the local ixTheo classification scheme.
+ *  \author  Dr. Johannes Ruscheinski
+ */
+
+/*
+    Copyright (C) 2015, Library of the University of Tübingen
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as
+    published by the Free Software Foundation, either version 3 of the
+    License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <iostream>
+#include <memory>
+#include <unordered_map>
+#include <cctype>
+#include <cstdlib>
+#include "DirectoryEntry.h"
+#include "Leader.h"
+#include "MarcUtil.h"
+#include "Subfields.h"
+#include "util.h"
+
+
+void Usage() {
+    std::cerr << "Usage: " << progname << "marc_input code_to_description_map\n";
+    std::exit(EXIT_FAILURE);
+}
+
+
+void LoadCodeToDescriptionMap(const std::shared_ptr<FILE> &code_to_description_map_file,
```

Comment:
why do you use this list? Why didn't you simplify the syntax of the list, something like
AB=Religionswissenschaft
The parsing of this function is very confusing.